src/Semantics.js

```diff
@@ -246,6 +310,42 @@ function parsePrototype(nameAndFormalArgs, allowFormals) {
   return prototypeGrammarSemantics(r).parse();
 }
 
+function newDefaultAction(typePlural, type, name, doIt) {
```

Comment:
I think it would be slightly clearer to just pass in the Operation/Attribute here, instead of `typePlural`.